src/verify_cert.rs

```diff
@@ -160,6 +172,100 @@ impl<'a> ChainOptions<'a> {
     }
 }
 
+/// Path from end-entity certificate to trust anchor that's been verified.
+///
+/// See [`EndEntityCert::verify_for_usage()`] for more details on what verification entails.
+pub struct VerifiedPath<'p> {
```

Comment:
do we want any derives on this? (i guess not, based on `EndEntityCert` and `Cert` not having any?)

Reply:
We should maybe reconsider that, but given that `EndEntityCert` and `Cert` don't have any it seems mostly orthogonal to this PR.